code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -1817,4 +1866,33 @@ impl<T: Config> Pallet<T> {
 			})
 			.collect::<Result<BTreeMap<_, _>, _>>()
 	}
+
+	fn allocate_shares(
+		pool_id: &AssetIdOf<T>,
+		fnft_account: &AccountIdOf<T>,
+		rewards_pool: &RewardPoolOf<T>,
+		awarded_shares: T::Balance,
+	) -> DispatchResult {
+		let pool_account = Self::pool_account_id(pool_id);
+
+		let unstaked_shares = T::Assets::balance(rewards_pool.share_asset_id, &pool_account);
+
+		let amount_to_transfer = match awarded_shares.checked_sub(&unstaked_shares) {
+			Some(amount_to_mint) => {
+				Self::mint_shares(rewards_pool.share_asset_id, amount_to_mint, fnft_account)?;
+				awarded_shares.defensive_saturating_sub(amount_to_mint)
+			},
+			None => awarded_shares,
+		};
+
+		T::Assets::transfer(
+			rewards_pool.share_asset_id,
+			&pool_account,
+			fnft_account,
+			amount_to_transfer,
+			false,
```

Comment:
```suggestion
			true,
```
Just to keep the dust amount safe

Reply:
This is being addressed in #2954, it's fine for now since I believe the ED is zero for the share assets. Setting this to true will cause an error here so probably ideal until #2954 is merged (we also handle this there)